backend/coins/btc/transaction.go

```diff
@@ -23,33 +25,56 @@ import (
 	"github.com/digitalbitbox/bitbox-wallet-app/backend/coins/btc/blockchain"
 	"github.com/digitalbitbox/bitbox-wallet-app/backend/coins/btc/maketx"
 	"github.com/digitalbitbox/bitbox-wallet-app/backend/coins/btc/transactions"
+	"github.com/digitalbitbox/bitbox-wallet-app/backend/coins/coin"
 	"github.com/digitalbitbox/bitbox-wallet-app/util/errp"
 )
 
+// unitSatoshi is 1 BTC (default unit) in Satoshi.
+const unitSatoshi = 1e8
+
 // TxValidationError represents errors in the tx proposal input data.
 type TxValidationError string
 
 func (err TxValidationError) Error() string {
 	return string(err)
 }
 
-// SendAmount is either a concrete amount, or "all"/"max".
+// SendAmount is either a concrete amount, or "all"/"max". The concrete amount is user input and is
+// parsed/validated in Amount().
 type SendAmount struct {
-	amount  btcutil.Amount
+	amount  string
 	sendAll bool
 }
 
 // NewSendAmount creates a new SendAmount based on a concrete amount.
-func NewSendAmount(amount btcutil.Amount) (SendAmount, error) {
-	if amount <= 0 {
-		return SendAmount{}, errp.New("invalid amount")
-	}
-	return SendAmount{amount: amount, sendAll: false}, nil
+func NewSendAmount(amount string) SendAmount {
+	return SendAmount{amount: amount, sendAll: false}
 }
 
 // NewSendAmountAll creates a new Sendall-amount.
 func NewSendAmountAll() SendAmount {
-	return SendAmount{amount: 0, sendAll: true}
+	return SendAmount{amount: "", sendAll: true}
+}
+
+// Amount parses the amount and converts it from the default unit to the smallest unit (e.g. satoshi
+// = 1e8). Returns an error if the amount is not positive.
+func (sendAmount *SendAmount) Amount(unit *big.Int) (coin.Amount, error) {
+	if sendAmount.sendAll {
+		panic("can only be called if SendAll is false")
+	}
+	amount, err := coin.NewAmountFromString(sendAmount.amount, unit)
+	if err != nil {
+		return coin.Amount{}, errp.WithStack(TxValidationError("invalid amount"))
+	}
+	if amount.Int().Cmp(big.NewInt(0)) <= 0 {
```

Comment:
You could also extract this into a helper method to avoid exposing big int, something like:

```go
func (a *Amount) Negative() bool {
    return a.amount.Sign() == -1
}

func (a *Amount) Zero() bool {
  return a.amount.Sign() == 0
}
```

And then you'd use it like:

```go
if amount.Negative() || amount.Zero() {
  return ..., err
}
```

Reply:
Also, you want to go without helpers, you could still simplify this with `amount.Sign() <= 0`, right?

https://golang.org/pkg/math/big/#Int.Sign

Reply:
Right! Done.